server/src/main/java/org/server/SupServer.java

```diff
@@ -112,4 +131,78 @@ public void tellSomeone(String message, String fromname, String toname) {
             } catch (Exception ex) { ex.printStackTrace(); }
         }
     }
-}
\ No newline at end of file
+    
+    /**
+     * return the status to the specific one
+	 *
+	 * @param 
+	 * 	status - status info
+	 *  toname - the specific one to transfer status
+     * */
+    public void tellSomeoneStatus(String status, String toname)
+    {
+    	System.out.println("Return the \"" + status +"\" to " + toname);
+    	if (Contacts.getInstance().hasContact(toname)) {
+            try {
+                PrintWriter writer = Contacts.getInstance().getContact(toname);
+                writer.println(status);
+                writer.flush();
+            } catch (Exception ex) { ex.printStackTrace(); }
+        }
+    }
+    
+    /**
+     * return the status to everyone
+	 *
+	 * @param 
+	 * 	status - status info
+     * */
+	public void tellEveryoneStatus(String status) {
```

Comment:
I know I just mentioned that in real life here, i just add it here for the sake of carrying on a conversation that everyone can participate in